gslab_scons/tests/test_build_r.py

```diff
@@ -4,35 +4,124 @@
+        in a standard way.
+        '''
+        mock_system.side_effect = self.os_system_side_effect
+        gs.build_r(target = './build/r.rds', 
+                   source = './input/script.R', 
+                   env    = '')
+        self.check_log('./build/sconscript.log')
+
+    @staticmethod
+    def os_system_side_effect(*args, **kwargs):
+        '''
+        This side effect mocks the behaviour of a system call
+        on a machine with R set up for command-line use.
+        '''
+        # Get and parse the command passed to os.system()
+        command = args[0]
+        match = re.match('\s*'
+                            '(?P<executable>R CMD BATCH)'
+                            '\s+'
+                            '(?P<option>--no-save)?'
+                            '\s*'
+                            '(?P<source>[\.\/\w]+(\.\w+)?)?'
+                            '\s*'
+                            '(?P<log>[\.\/\w]+(\.\w+)?)?',
+                         command)
+
+        executable = match.group('executable')
+        log        = match.group('log')
+
+        # As long as the executable is correct and a log path 
+        # is specified, write a log.
+        if executable == "R CMD BATCH" and log:
+            with open(log.strip(), 'wb') as log_file:
+                log_file.write('Test log\n')
+
+    def check_log(self, log_path = './build/sconscript.log'):
+        '''
+        Check that log_path is a file that has a 
+        log-creation timestamp.
+        '''
+        self.assertTrue(os.path.isfile(log_path))
+        
+        with open(log_path, 'rU') as log_file:
+            log_data = log_file.read()
+        self.assertIn('Log created:', log_data)
+
+        os.remove(log_path)
+
+    @mock.patch('gslab_scons.builders.build_r.os.system')
+    def test_target_list(self, mock_system):
+        mock_system.side_effect = self.os_system_side_effect
+        # We don't expect that the targets actually need
+        # to be created.
+        targets = ['./build/r.rds', 'additional_target']
+        gs.build_r(target = targets, 
+                   source = './script.R', 
+                   env    = '')
+        # We expect build_r() to write its log to its 
+        # first target's directory
+        self.check_log('./build/sconscript.log')
 
     def test_bad_extension(self):
         '''Test that build_r() recognises an inappropriate file extension'''
-        env = ''
-        with self.assertRaises(BadExtensionError):
-            build_r('../build/r.rds', ['bad', './input/R_test_script.R'], env)
+        with self.assertRaises(BadExtensionError), nostderrout():
+            gs.build_r(target = './build/r.rds', 
+                       source = ['bad', './script.R'], 
+                       env    = '')
+
+    @mock.patch('gslab_scons.builders.build_r.os.system')
+    def test_unintended_inputs(self, mock_system):
+        # We don't expect the env argument to affect the function's
```

Comment:
OK. I will pull in the current master and update the tests as is appropriate unless you object.